netty/src/main/java/io/grpc/netty/NettyClientHandler.java

```diff
@@ -358,27 +370,27 @@ public void operationComplete(ChannelFuture future) throws Exception {
                   // The http2Stream will be null in case a stream buffered in the encoder
                   // was canceled via RST_STREAM.
                   Http2Stream http2Stream = connection().stream(streamId);
-                  if (http2Stream != null) {
+                  if (http2Stream != null && !stream.isClosed()) {
```

Comment:
Why the `!stream.isClosed()`? That seems like it can cause trouble if the stream was cancelled by the application but netty hasn't yet sent the RST_STREAM.